### DISCUSSION TITLE
How to replace Bentley authentication with custom authentication in iTWIN ?

### DISCUSSION THREAD

**Discussion by nawazmujawar1** [iTwin/itwinjs-core#2539] [Q&A] [answered]
I want to replace the authentication part given in iTWIN template project with my custom one. So that the user logins through custom login page and then he can see the Viewer.

**Comment by calebmshafer** [comment, marked as answer]
Hey @nawazmujawar1, I wrote up a sample that shows how to replace the Bentley identity provider with your own and leverage the iTwin Platform. Here's the link to the [repo](https://github.com/calebmshafer/itwin-viewer-proxy/), I'd recommend following approach #2 in the README.

Also, if your own identity provider is using OIDC/OAuth2 you can still use our BrowserAuthorizationClient with different configurations to do the initial sign-in within the viewer. (That is before calling to the token server to get a valid Bentley access token).

Let me know if anything is unclear in the sample.
